### ISSUE TITLE
Realtime queries are not completely filtered

### ISSUE THREAD

**Issue by Sn0wfreezeDev** [jsayol/FireSQL#3]
Hi, 

I tested out your SQL library for Firebase CloudFirestore and I really like it!
It's great work and I appreciate the development of it. Thank you. 

I found a small issue that updates received with realtime subscriptions are not filtered. 

Here's my example: 
I make an SQL query with an OR to compare to multiple features. Then I use sqlRxQuery to subscribe to realtime updates. 

SQL Query:
```SQL
SELECT * FROM venues WHERE `clusters.cid1` = true OR `clusters.cid2` = true
```

JS Code:
```javascript
firesql.rxQuery(queryString)
     .subscribe(documents => {
       console.log("Received %d venues in the clusters", documents.length)
       let uniqueDocs =  unique(documents);
       console.log("Unique documents %d", uniqueDocs.length)
       state.venues = uniqueDocs;
     })

function unique(a) {
  var seen = {};
  return a.filter(function(item) {
      return seen.hasOwnProperty(item.id) ? false : (seen[item.id] = true);
  });
}


```

Logs:
```Log
Received 143 venues in the clusters
Unique documents 104
```

Thank you a lot for implementing this

**Comment by jsayol** [comment]
Thanks for the detailed report! You're absolutely right, `rxQuery()` wasn't filtering duplicate documents when combining multiple queries.

I just published a new version (`1.0.1`) fixing this issue.

Thanks!

**Comment by Sn0wfreezeDev** [comment]
Wow thank you for the quick fix!
